rate/fetchers/base.go

```diff
@@ -0,0 +1,47 @@
+package fetchers
```

Comment:
Why do you need this base handler? It's not present even in refactor guru example. I think you can remove it and your code will behave the same way. 
https://refactoring.guru/design-patterns/chain-of-responsibility/go/example